src/utils/mergeTree.js

```diff
@@ -212,37 +233,53 @@ async function mergeBlobs({
   const type = 'blob'
   // Compute the new mode.
   // Since there are ONLY two valid blob modes ('100755' and '100644') it boils down to this
-  const mode =
-    (await base.mode()) === (await ours.mode())
-      ? await theirs.mode()
-      : await ours.mode()
-  // The trivial case: nothing to merge except maybe mode
-  if ((await ours.oid()) === (await theirs.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
-  }
-  // if only one side made oid changes, return that side's oid
-  if ((await ours.oid()) === (await base.oid())) {
-    return { mode, path, oid: await theirs.oid(), type }
-  }
-  if ((await theirs.oid()) === (await base.oid())) {
-    return { mode, path, oid: await ours.oid(), type }
+  let ourMode = await ours.mode();
+  if (base !== null) {
+    const mode =
+      (await base.mode()) === (ourMode)
+        ? await theirs.mode()
+        : ourMode
+    // The trivial case: nothing to merge except maybe mode
+    if ((await ours.oid()) === (await theirs.oid())) {
+      return { mode, path, oid: await ours.oid(), type }
+    }
+    // if only one side made oid changes, return that side's oid
+    if ((await ours.oid()) === (await base.oid())) {
+      return { mode, path, oid: await theirs.oid(), type }
+    }
+    if ((await theirs.oid()) === (await base.oid())) {
+      return { mode, path, oid: await ours.oid(), type }
+    }
   }
   // if both sides made changes do a merge
+  let baseContent;
+  try {
+    baseContent = Buffer.from(await base.content()).toString('utf8');
+  } catch(error) {
+    baseContent = Buffer.from("").toString('utf8');
```

Comment:
This is just an empty string. If you assign `baseContent` to the empty string above, the `catch` doesn't need to do anything.